### PR TITLE
feat: badge for SwiftUI

### PR DESCRIPTION
**Pull request by florentlotthepro**
## Changes description

I added the badge component for SwiftUI.

## Context

The component was done for UIKit, now it's also done for SwiftUI.

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on an iPhone device/simulator.
- [x] I have tested on an iPad device/simulator.
- [x] If it includes design changes, please ask for a review with a core team designer.

## Does this introduce a breaking change?

- No

#### iPhone

Simulator Screen Shot - iPhone 14 - 2022-10-28 at 18 34 00
Simulator Screen Shot - iPhone 14 - 2022-10-28 at 18 34 03
Simulator Screen Shot - iPhone 14 - 2022-10-28 at 18 34 06

#### iPad

Simulator Screen Shot - iPad Pro (9 7-inch) - 2022-10-28 at 18 40 29
Simulator Screen Shot - iPad Pro (9 7-inch) - 2022-10-28 at 18 40 32

## Other information

Like it was done on UIKit, I also provide a method to add a badge to the top right of a view.

### CODE REVIEW

File: Showcase/Application/Core/Components/Badge/BadgeModel.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    struct Item: Identifiable {
+        var id: String {
+            "\(value ?? 0)-\(variant.name)"
+        }
+        var value: Int?
+        var variant: VitaminBadgeVariant = .standard
+    }
+
+    static let sections: [BadgeModel.Section] = {
+        var varientSection: [BadgeModel.Section] = []
```
Comment (daniel-dumortier) [inline Showcase/Application/Core/Components/Badge/BadgeModel.swift:26]: Just a little typo....
```suggestion
        var variantSection: [BadgeModel.Section] = []
```

File: Showcase/Application/Core/Components/Badge/BadgeModel.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        for variant in VitaminBadgeVariant.allCases {
+            let section = BadgeModel.Section(name: variant.name, items: [
+                BadgeModel.Item(value: nil, variant: variant),
+                BadgeModel.Item(value: 2, variant: variant),
+                BadgeModel.Item(value: 50, variant: variant),
+                BadgeModel.Item(value: 100, variant: variant)
+            ])
+            varientSection.append(section)
```
Comment (daniel-dumortier) [inline Showcase/Application/Core/Components/Badge/BadgeModel.swift:34]:
```suggestion
            variantSection.append(section)
```

File: Sources/VitaminCore/Foundations/Typography/TextStyles.swift [hunk 3]
```diff
@@ -198,6 +208,13 @@ extension VitaminTextStyle {
         (lineHeight * font.pointSize) / size
     }
 
+    /// The scaled line height with the provided font for the current style.
+    /// - Parameter font: The font for which we want the line height.
```
Comment (daniel-dumortier) [inline Sources/VitaminCore/Foundations/Typography/TextStyles.swift:212]: The function takes no parameter

**Review by daniel-dumortier — APPROVED** [review]
Great job ! 👍
Just two minor remarks.

File: Showcase/Application/Core/Components/Badge/BadgeModel.swift (hunk 1 above)
Comment (florentlotthepro) [reply Showcase/Application/Core/Components/Badge/BadgeModel.swift:26]: 🙈

**Review by baptistedajon — APPROVED** [review]

**Review by daniel-dumortier — APPROVED** [review]
👍

**Pull request merged**